routing/routing_tests/index_graph_tools.cpp

```diff
@@ -41,6 +46,150 @@ void TestIndexGraphLoader::AddGraph(NumMwmId mwmId, unique_ptr<IndexGraph> graph
+{
+  m_edgeRequests.emplace_back(vertexId1, vertexId2, weight);
+}
+
+bool TestIndexGraphTopology::FindPath(uint32_t start, uint32_t finish, double & pathWeight,
+                                      vector<pair<uint32_t, uint32_t>> & pathEdges)
+{
+  // Edges of the index graph are segments, so we need a loop at finish
+  // for the end of our path and another loop at start for the bidirectional search.
+  AddDirectedEdge(start, start, 0.0);
+  AddDirectedEdge(finish, finish, 0.0);
+  MY_SCOPE_GUARD(cleanup, [&]() {
+    m_edgeRequests.pop_back();
+    m_edgeRequests.pop_back();
+  });
+
+  BuildGraphFromRequests();
+  auto const worldGraph = PrepareIndexGraph();
+  CHECK(worldGraph != nullptr, ());
+
+  if (m_joints[start].GetSize() == 0 || m_joints[finish].GetSize() == 0)
+  {
+    pathWeight = 0;
+    pathEdges.clear();
+    return false;
+  }
+
+  auto const startRoadPoint = m_joints[start].GetEntry(0);
+  auto const finishRoadPoint = m_joints[finish].GetEntry(0);
+
+  auto const fakeStart =
+      IndexGraphStarter::FakeVertex(kTestNumMwmId, startRoadPoint.GetFeatureId(),
+                                    startRoadPoint.GetPointId(), m2::PointD::Zero());
+  auto const fakeFinish =
+      IndexGraphStarter::FakeVertex(kTestNumMwmId, finishRoadPoint.GetFeatureId(),
+                                    finishRoadPoint.GetPointId(), m2::PointD::Zero());
+
+  IndexGraphStarter starter(fakeStart, fakeFinish, *worldGraph);
+
+  vector<Segment> routeSegs;
+  double timeSec;
+  auto const resultCode = CalculateRoute(starter, routeSegs, timeSec);
+
+  if (resultCode == AStarAlgorithm<IndexGraphStarter>::Result::NoPath)
+    return false;
+  CHECK_EQUAL(resultCode, AStarAlgorithm<IndexGraphStarter>::Result::OK, ());
+
+  CHECK_GREATER_OR_EQUAL(routeSegs.size(), 2, ());
+  CHECK_EQUAL(routeSegs.front(), starter.GetStart(), ());
+  CHECK_EQUAL(routeSegs.back(), starter.GetFinish(), ());
+  // We are not interested in fake start and finish.
+  pathEdges.resize(routeSegs.size() - 2);
+  pathWeight = 0;
+  for (size_t i = 1; i + 1 < routeSegs.size(); ++i)
+  {
+    auto const & seg = routeSegs[i];
+    auto const it = m_segmentToEdge.find(seg);
+    CHECK(it != m_segmentToEdge.cend(), ());
+    auto const & edge = it->second;
+    pathEdges[i - 1] = edge;
+    pathWeight += m_segmentWeights[seg];
+  }
+
+  CHECK(!pathEdges.empty(), ());
+  // The loop from finish to finish.
+  pathEdges.pop_back();
+
+  return true;
+}
+
+unique_ptr<WorldGraph> TestIndexGraphTopology::PrepareIndexGraph()
+{
+  unique_ptr<TestGeometryLoader> loader = make_unique<TestGeometryLoader>();
+
+  for (auto const & p : m_segmentWeights)
+  {
+    double const weight = p.second;
+    loader->AddRoad(p.first.GetFeatureId(), true /* oneWay */, 1.0 /* speed */,
+                    RoadGeometry::Points({{0.0, 0.0}, {weight, 0.0}}));
+  }
+
+  BuildJoints();
+
+  shared_ptr<EdgeEstimator> estimator =
+      EdgeEstimator::CreateForWeightedDirectedGraph(m_segmentWeights);
```

Comment:
You use CreateForWeightedDirectedGraph only in tests. So, why do you need CreateForWeightedDirectedGraph and WeightedDirectedGraphEdgeEstimator in the routing directory? Why not to move them to the tests directory?